libs/testserver/server.go

```diff
@@ -2,34 +2,24 @@ package testserver
 
 import (
 	"encoding/json"
-	"fmt"
 	"io"
 	"net/http"
-	"net/http/httptest"
 	"net/url"
 	"reflect"
 	"strings"
-	"sync"
-
-	"github.com/gorilla/mux"
 
 	"github.com/databricks/cli/internal/testutil"
-	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/gorilla/mux"
 )
 
-type Server struct {
```

Comment:
Rather than refactoring libs/testserver, how about keeping it as is and adding libs/testproxy for proxy functionality (it can depend on libs/testserver for auxiliary classes such Request and EncodedResponse.

This would keep testserver simple and changes in it minimal, making the whole thing easier to review.

Reply:
I approached it this way because it makes it easier to add common functionality in both servers in the future. It keeps test servers easier to reason about. The refactor is pretty safe since the existing tests pass.

Reply:
moved proxy server to a separate package.